packages/terra-application/src/notification-dialog/NotificationDialog.jsx

```diff
@@ -0,0 +1,289 @@
+        const tabbableElements = tabbable(containerElement);
+        if (!tabbableElements.length) {
+          return;
+        }
+
+        const indexOfActiveElement = tabbableElements.indexOf(document.activeElement);
+        let newActiveElementIndex;
+        if (event.shiftKey) {
+          if (indexOfActiveElement === -1) {
+            newActiveElementIndex = tabbableElements.length - 1;
+          } else {
+            newActiveElementIndex = indexOfActiveElement === 0
+              ? tabbableElements.length - 1 : indexOfActiveElement - 1;
+          }
+        } else if (indexOfActiveElement === -1) {
+          newActiveElementIndex = 0;
+        } else {
+          newActiveElementIndex = indexOfActiveElement === tabbableElements.length - 1
+            ? 0 : indexOfActiveElement + 1;
+        }
+
+        tabbableElements[newActiveElementIndex].focus();
+        event.preventDefault();
+      }
+    };
+
+    containerElement.addEventListener('keydown', moveFocus);
+
+    return () => {
+      containerElement.removeEventListener('keydown', moveFocus);
+    };
+  }, [containerRef]);
+};
+
+const NotificationDialog = ({
+  dialogTitle,
+  startMessage,
+  endMessage,
+  content,
+  acceptAction,
+  rejectAction,
+  variant,
+  buttonOrder,
+  emphasizedAction,
+  custom,
+}) => {
+  const theme = React.useContext(ThemeContext);
+  const applicationIntl = React.useContext(ApplicationIntlContext);
+  const { LayerPortal, layerId } = useLayerPortal({ layerType: 'notificationDialog' });
+  const dialogContainerRef = React.useRef();
+  const initialFocusAnchorRef = React.useRef();
+
+  useFocusTrap(dialogContainerRef);
+
+  React.useEffect(() => {
+    deferExecution(() => {
+      if (initialFocusAnchorRef.current) {
+        initialFocusAnchorRef.current.focus();
+      }
+    });
+  }, []);
+
+  if (acceptAction === undefined && rejectAction === undefined) {
+    throw new Error('[terra-application] Either the `acceptAction` or `rejectAction` props must be provided to the NotificationDialog');
+  }
+
+  if (variant === undefined) {
+    throw new Error('[terra-application] A variant must be provided to the NotificationDialog');
+  }
+
+  const signalWord = variant === 'custom'
+    ? custom.signalWord : applicationIntl.formatMessage({
+      id: `terraApplication.notificationDialog.${variant}`,
+    });
+
+  const bodyContentId = `${layerId}-body`;
+  const actionsContentId = `${layerId}-actions`;
+
+  return (
+    <LayerPortal>
+      <div
+        className={cx('presentation-container', theme.className)}
+        tabIndex="-1"
+        role="alertdialog"
+        aria-label={`${signalWord} ${dialogTitle}`}
+        aria-describedby={`${bodyContentId} ${actionsContentId}`}
+        aria-modal="true"
+        ref={dialogContainerRef}
+      >
+        <div className={cx('overlay')} />
+        <div className={cx('notification-dialog')}>
+          <div tabIndex="-1" ref={initialFocusAnchorRef} data-testid="dialog-focus-anchor" />
```

Comment:
Does this get focus from a screen reader, or is prevented in some manor?

Reply:
It only gets focus when the dialog initially opens. It's otherwise inaccessible, at least to VoiceOver. I couldn't put focus on the alertdialog element itself because it reads the content twice. Putting focus here on an element with no text causes the reader to bubble up to the next element and makes it read appropriately the single time.

Reply:
Cool cool, down the line we might verify with JAWS as well, but that makes sense to me.